### ISSUE TITLE
v4l2rtspserver-master out of date, causing video settings to not apply properly

### ISSUE THREAD

**Issue by aartrost** [EliasKotlyar/Xiaomi-Dafang-Hacks#459]
Hello all,

First of all: thank you to all the contributors for this awesome piece of software. I love having a surveillance system that's fully under my control especially with all the privacy violations going on recently.

I spent a lot of time attempting to tune the video settings on my Dafang camera using the latest version of this repository. The video settings do not seem to apply because the current build of v4l2rtspserver is out of date, and does not contain the right config endpoints for the available video settings in the administration panel. Setting the video format to anything other than FixedQp breaks and the bitrate setting also does not work with this outdated build. I spent some time figuring out how to build v4l2rtspserver-master from source, and this seems to fix all issues, allowing me to get MUCH better video quality with VBR!  

I'm not intimately familiar with this project, so I'm hesitant to simply commit my build of v4l2rtspserver-master into the repo. However, if desired I can create a pull request.

**Comment by jmtatsch** [comment]
Did you build from source with the upstream repo, our repo or some sdk source?

**Comment by aartrost** [comment]
Hi @jmtatsch!

I built v4l2rtspserver from the Dafang-Hacks repo: https://github.com/Dafang-Hacks/Main.

**Comment by nik0** [comment]
Hi, 
It would be nice if you can create a PR .../

**Comment by aartrost** [comment]
Sure thing! Find the PR here: https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/pull/460

**Comment by nik0** [comment]
Can you also push the code you modified in the server ?

**Comment by aartrost** [comment]
Hey @nik0, 

I didn't modify any code, just built the latest version!

**Comment by nik0** [comment]
OK ... 
So I don't understand why the video settings are better
The difference between previous version is the way the detection script are started. 
Maybe @EliasKotlyar changed the compiler

**Comment by aartrost** [comment]
Yeah, I'm not 100% sure either. I received my Dafang cameras 2 days (so I couldn't be on an older version of the codebase) ago and the rtsp stream would crash if I changed the video format to anything other than FixedQp. Changing bitrate didn't do anything either. Only after rebuilding the v4l2rtspserver binary this all started working again. I reproduced this behavior on 2 separate cameras.

**Comment by EliasKotlyar** [comment]
@nik0 : I havent changed the compiler yet. I can imagine that the change of the libimp binaries make a difference....

**Comment by nik0** [comment]
I created a new PR with the last binary. Tell me if it OK for you and everything is working as expected

**Comment by saipsa** [comment]
@aartrost 

>  allowing me to get MUCH better video quality with VBR!

Can you please share your configuration? Thanks.

**Comment by aartrost** [comment]
Hi @saipsa,

I have video size set to 1280 x 720, bitrate at 5000 and Video Format set to VBR. The key was to use the new v4l2rtspserver-master binary. I see that this new binary has been recently pushed to the repository, so make sure you update the camera to the latest version. Let me know if that worked! I haven't had time to test the latest commits myself.

**Comment by saipsa** [comment]
@aartrost 

Thanks.

I tried that. Moving objects leave behind pixelated trail. at that bit rate. maximum bitrate without noticable trail seems 2000 for me. Other settings are same as yours.

I think by reducing fps we can increase the resolution further.

**Comment by aartrost** [comment]
Odd, I have no problems pushing the bitrate/resolution up. I only start seeing artifacts after a bitrate of ~6500. I do have this special bootloader installed: https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/issues/30#issuecomment-354625120, perhaps this might improve performance for lower resolutions as well?

**Comment by jmtatsch** [comment]
Is this solved now?

**Comment by aartrost** [comment]
It is! Closing now.
